bugsnag/client.py

```diff
@@ -117,12 +122,22 @@ def deliver(self, notification):  # type: (Notification) -> None
         if not self.should_deliver(notification):
             return
 
+        initial_severity = notification.severity
+        initial_reason = notification.severity_reason
+
         def send_payload():
             if notification.api_key is None:
                 bugsnag.logger.warning(
                     "No API key configured, couldn't notify")
                 return
 
+            if initial_severity != notification.severity:
+                notification.severity_reason = {
+                    'type': 'userCallbackSetSeverity'
+                }
+            else:
+                notification.severity_reason = initial_reason
```

Comment:
Why do we do this here?

Reply:
This enforces that users cannot overwrite the severity_reason in callbacks or middleware.  I think we previously said it wouldn't matter if someone chose to do so, but we've added it as one of our testing parameters and are enforcing the same thing across notifiers.